recipe/meta.yaml

```diff
@@ -17,17 +17,16 @@ build:
 
 requirements:
   host:
-    - python
+    - python >=3.7
     - pip
     - wheel
     - setuptools >=30.0.0
   run:
-    - python >=3.6
-    # mccabe 0.7.0 is a noarch python package now and it hasn't broken changes for flake8
-    - mccabe >=0.6.0,<0.8.0
+    - python >=3.7
+    - mccabe >=0.6.0,<0.7.0
     - pycodestyle >=2.8.0,<2.9.0
     - pyflakes >=2.4.0,<2.5.0
-    - importlib-metadata
+    - importlib-metadata <4.3  # [py<38]
```

Comment:
This restriction + selector is not going to do anything if we're keeping `flake8` a `noarch: python` package. If a limit on `importlib-metadata` is all we need, a [repodata patch](https://github.com/AnacondaRecipes/repodata-hotfixes/) is probably the better approach.